aten/src/ATen/native/cudnn/RNN.cpp

```diff
@@ -378,6 +379,62 @@ namespace {
     }
   }
 
+  void add_projection_weights(
+        cudnnHandle_t handle,
+        const RNNDescriptor& rnn_desc,
+        const TensorDescriptor& x_desc,
+        const FilterDescriptor& w_desc,
+        const Tensor& weight_buf,
+        int64_t layer,
+        std::vector<Tensor>& params
+  ) {
+    void* matrix_pointer = nullptr;
+    // assuming it's LSTM which has 8 layers
+    int64_t linear_id = 8;
```

Comment:
I agree that calling this "layers" is misleading. But I was just following the convention used in other parts of the code, e.g. "_num_linear_layers" function defined in this file or "num_linear_layers" variable used in "get_parameters" function. Let me explain this in the comment here, so that it's clearer. For 8 as projection weight, I actually took it from TensorFlow implementation for CuDNN LSTM. Not sure if this is the best way to do it, but it seems to be working.